bin/utest/signal.c

```diff
@@ -295,6 +296,114 @@ TEST_ADD(signal_sigsuspend) {
   return 0;
 }
 
+/* ======= signal_sigtimedwait ======= */
+int sigtimedwait_child(void *arg) {
+  pid_t ppid = getppid();
+  syscall_ok(kill(ppid, SIGUSR1));
+  return 0;
+}
+
+TEST_ADD(signal_sigtimedwait) {
+  syscall_ok(signal(SIGCONT, sigcont_handler));
+  sigset_t set, current, waitset;
+  __sigemptyset(&set);
+  __sigaddset(&set, SIGUSR1);
+  __sigaddset(&set, SIGCONT);
```

Comment:
s/__sig...set/sig...set/

see /lib/libc/gen/sig...set.c

Reply:
Indeed `__sig*` is not a part of POSIX interface, please have a look at [sigsetops](https://man.netbsd.org/sigsetops.3). The issue here is that, nonetheless, those functions are exposed to non-kernel code, instead of being hidden as in FreeBSD: https://mimiker.ii.uni.wroc.pl/source/xref/FreeBSD/lib/libc/gen/sigsetops.c?r=7a56009c#40